src/math/trig.c

```diff
@@ -5,98 +5,212 @@
- *	pi/2 is added or subtracted from the angle until it is within the range
- *	[-pi/2,pi/2).Start with the angle in the range [-2*pi, 2*pi) and output
- *	has range in [-1.0 to 1.0]
- *	+------------------+-----------------+--------+--------+
- *	| thRadFxp	   | cdcsinth	     |thRadFxp|cdcsinth|
- *	+----+-----+-------+----+----+-------+--------+--------+
- *	|WLen| FLen|Signbit|WLen|FLen|Signbit| Qformat| Qformat|
- *	+----+-----+-------+----+----+-------+--------+--------+
- *	| 32 | 28  |  1	   | 32	| 31 |	 1   | 4.28   | 1.31   |
- *	+------------------+-----------------+--------+--------+
- */
-
+#ifdef CONFIG_CORDIC_TRIGONOMETRY_FIXED
 /* Use a local definition to avoid adding a dependency on <math.h> */
 #define _M_PI		3.14159265358979323846	/* pi */
-
-int32_t sin_fixed(int32_t th_rad_fxp)
-{
-	/*cordic_atan2_lookup_table = atan(2.^-(0:N-1)) N = 31
-	 *CORDIC Gain is cordic_gain = prod(sqrt(1 + 2.^(-2*(0:31-1))))
-	 *Inverse CORDIC Gain,inverse_cordic_gain = 1 / cordic_gain
-	 */
-	static const int32_t cordicsine_lookup[CORDICSINE_TABLE_SIZE] = { 843314857, 497837829,
+/*cordic_atan2_lookup_table = atan(2.^-(0:N-1)) N = 31/16
+ *CORDIC Gain is cordic_gain = prod(sqrt(1 + 2.^(-2*(0:31/16-1))))
+ *Inverse CORDIC Gain,inverse_cordic_gain = 1 / cordic_gain
+ */
+static const int32_t cordic_lookup[CORDIC_31B_TABLE_SIZE] = { 843314857, 497837829,
 	263043837, 133525159, 67021687, 33543516, 16775851, 8388437, 4194283, 2097149,
 	1048576, 524288, 262144, 131072, 65536, 32768, 16384, 8192, 4096, 2048, 1024,
 	512, 256, 128, 64, 32, 16, 8, 4, 2, 1 };
-	int32_t b_idx;
+
+/* 652032874 , deg = 69.586061*/
+const int32_t cordic_sine_cos_lut_q29fl	 =  Q_CONVERT_FLOAT(1.214505869895220, 29);
+/* 1686629713, deg = 90.000000	*/
+const int32_t cordic_sine_cos_piovertwo_q30fl  = Q_CONVERT_FLOAT(_M_PI / 2, 30);
+/* 421657428 , deg = 90.000000 */
+const int32_t cord_sincos_piovertwo_q28fl  = Q_CONVERT_FLOAT(_M_PI / 2, 28);
+/* 843314857,  deg = 90.000000	*/
+const int32_t cord_sincos_piovertwo_q29fl  = Q_CONVERT_FLOAT(_M_PI / 2, 29);
+
+/**
+ * \Compute fixed point cordicsine with table lookup and interpolation
+ * \The cordic sine algorithm converges, when the angle is in the range
+ * \[-pi/2, pi/2).If an angle is outside of this range, then a multiple of
+ * \pi/2 is added or subtracted from the angle until it is within the range
+ * \[-pi/2,pi/2).Start with the angle in the range [-2*pi, 2*pi) and output
+ * \has range in [-1.0 to 1.0]
+ * \+------------------+-----------------+--------+--------+
+ * \| thRadFxp	       | cdcsinth	 |thRadFxp|cdcsinth|
+ * \+----+-----+-------+----+----+-------+--------+--------+
+ * \|WLen| FLen|Signbit|WLen|FLen|Signbit| Qformat| Qformat|
+ * \+----+-----+-------+----+----+-------+--------+--------+
+ * \| 32 | 28  |  1    | 32 | 31 |   1	 | 4.28	  | 1.31   |
+ * \+------------------+-----------------+--------+--------+
+ */
+inline int32_t sin_fixed_32b(int32_t th_rad_fxp)
+{
+	int32_t sign;
```

Comment:
e.g. here, as this will generate 2 function calls when used outside of trig.c i.e.
1) sin_fixed_32b()
2) cordic_sin_cos()

We want to eliminate call 1.

Same for all the other small maths functions.

Reply:
https://github.com/thesofproject/sof/pull/4218